### DISCUSSION TITLE
Is it possible to replace the current leader master to another master?

### DISCUSSION THREAD

**Discussion by guol-fnst** [seaweedfs/seaweedfs#4078] [Q&A] [answered]
1) mark the current leader  to be unavailable and stop accept new requests
2) notify volume/filer the new leader master
3) start service with new leader
4) shutdown the older leader in order to  upgrage weed or migirate to a new server

If this is possible, we can upgrage seaweedfs cluster smoothly

**Comment by kmlebedev** [comment, marked as answer]
Leader transfer If you use hashicorp raft and  send HUP signal to the master leader.

**Reply by guol-fnst** [reply]
Got it, Thanks!
